src/common/dbg.c

```diff
@@ -43,15 +44,23 @@ static inline void _dbg_unlock(void) {
 
 void _dbg_swo(const char *fmt, ...) {
     _dbg_lock();
+
     char line[DBG_MAXLINE];
     va_list va;
     va_start(va, fmt);
     int len = vsnprintf(line, DBG_MAXLINE, fmt, va);
     va_end(va);
-    line[len++] = '\n';
-    for (int i = 0; i < len; i++) {
-        ITM_SendChar(line[i]);
+
+    if (len < 0) {
+        _dbg_unlock();
+        return;
     }
+    len = MIN(len, DBG_MAXLINE - 1);
```

Comment:
I'd like to see unit tests for these fixes so that no other buffer overrun can occur in the future